### PR TITLE
kvrpc: add raw scan.

### PR DESCRIPTION
**Pull request by disksing**
cc @AndreMouche @siddontang

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -453,6 +460,23 @@ fn process_read(cid: u64, mut cmd: Command, ch: SyncSendCh<Msg>, snapshot: Box<S
     }
 }
 
+fn process_rawscan(snapshot: Box<Snapshot>,
+                   start_key: &Key,
+                   limit: usize,
+                   stats: &mut Statistics)
+                   -> Result<Vec<StorageResult<KvPair>>> {
```
Comment (AndreMouche) [inline src/storage/txn/scheduler.rs]: It seems no error would been returned, why not return  `Vec<KvPair>` directly?

File: tests/storage/test_storage.rs [hunk 2]
```diff
@@ -428,6 +428,17 @@ fn test_txn_store_rawkv() {
...
+    store.raw_put_ok(b"k1".to_vec(), b"v1".to_vec());
+    store.raw_put_ok(b"k2".to_vec(), b"v2".to_vec());
+    store.raw_put_ok(b"k3".to_vec(), b"v3".to_vec());
+    store.raw_scan_ok(b"".to_vec(), 1, vec![(b"k1", b"v1")]);
+    store.raw_scan_ok(b"k1".to_vec(), 1, vec![(b"k1", b"v1")]);
+    store.raw_scan_ok(b"k10".to_vec(), 1, vec![(b"k2", b"v2")]);
+    store.raw_scan_ok(b"".to_vec(), 2, vec![(b"k1", b"v1"), (b"k2", b"v2")]);
+    store.raw_scan_ok(b"k1".to_vec(),
+                      5,
+                      vec![(b"k1", b"v1"), (b"k2", b"v2"), (b"k3", b"v3")]);
```
Comment (AndreMouche) [inline tests/storage/test_storage.rs]: Could we add test cases scan with `start_key=b"k5"` or `limit = 0`?

File: Cargo.lock [hunk 3]
```diff
@@ -15,7 +15,7 @@ dependencies = [
  "futures-cpupool 0.1.5 (registry+https://github.com/rust-lang/crates.io-index)",
  "grpc 0.1.0 (git+https://github.com/pingcap/grpc-rs.git)",
  "jemallocator 0.1.0 (git+https://github.com/busyjay/jemallocator.git?branch=dev)",
- "kvproto 0.0.1 (git+https://github.com/pingcap/kvproto.git)",
+ "kvproto 0.0.1 (git+https://github.com/pingcap/kvproto.git?branch=disksing/rawscan)",
```
Comment (AndreMouche) [inline Cargo.lock]: Please update kvproto since the related PR is already merged

File: src/storage/txn/scheduler.rs (hunk 1 above)
Comment (disksing) [inline src/storage/txn/scheduler.rs]: I want to reuse the ProcessResult here https://github.com/pingcap/tikv/blob/52693838cb0eec496d8c1268935bcea826844ce4/src/storage/txn/scheduler.rs#L69

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**